src/transformers/models/tvlt/modeling_tvlt.py

```diff
@@ -0,0 +1,1426 @@
+        self.pos_embed_a = nn.Parameter(torch.zeros(1, self.num_patches // self.num_freq_patches, config.hidden_size))
+        self.freq_embed = nn.Parameter(torch.zeros(1, self.num_freq_patches, config.hidden_size))
+
+        self.num_freq_patches = config.feature_size // config.audio_patch_size[1]
+        self.config = config
+
+    def random_masking(self, sequence, attention_masks=None, mask_ratio=0.15):
+        """
+        Perform random masking by per-sample shuffling on frame-level.
+        Per-sample shuffling is done by argsort random noise.
+        sequence: [batch_size, seq_len, hidden_dim], sequence
+        """
+
+        batch_size, seq_len, hidden_dim = sequence.shape
+        freq_len, time_len = self.num_freq_patches, seq_len // self.num_freq_patches
+
+        len_keep = int(seq_len * (1 - mask_ratio))
+        noise = (
+            torch.rand(batch_size, time_len, device=sequence.device)
+            .unsqueeze(-1)
+            .repeat(1, 1, freq_len)
+            .view(batch_size, seq_len)
+        )  # noise in [0, 1]
+
+        # sort noise for each sample
+        ids_shuffle = torch.argsort(noise, dim=1)  # ascend: small is keep, large is remove
+        ids_restore = torch.argsort(ids_shuffle, dim=1)
+
+        # keep the first subset
+        ids_keep = ids_shuffle[:, :len_keep]
+        sequence_masked = torch.gather(sequence, dim=1, index=ids_keep.unsqueeze(-1).repeat(1, 1, hidden_dim))
+
+        # generate the binary mask: 0 is keep, 1 is remove
+        label_masks = torch.ones([batch_size, seq_len], device=sequence.device)
+        label_masks[:, :len_keep] = 0
+        # unshuffle to get the binary mask
+        label_masks = torch.gather(label_masks, dim=1, index=ids_restore)
+
+        if attention_masks is not None:
+            label_masks *= attention_masks
+            attention_masks = torch.gather(attention_masks, dim=1, index=ids_keep)
+
+        return sequence_masked, attention_masks, label_masks, ids_restore
+
+    def forward(self, pixel_values, attention_masks=None, mask_ratio=0.0):
+        # create patch embeddings
+        embeddings = self.patch_embeddings(pixel_values)
+
+        num_time_patches = embeddings.size(1) // self.num_freq_patches
+        embeddings += self.freq_embed.repeat(1, num_time_patches, 1)
+        embeddings += torch.repeat_interleave(self.pos_embed_a[:, :num_time_patches], self.num_freq_patches, dim=1)
+        embeddings += self.type_embed_a
+
+        label_masks = None
+        ids_restore = None
+        if mask_ratio > 0.0:
+            embeddings, attention_masks, label_masks, ids_restore = self.random_masking(
+                embeddings, attention_masks=attention_masks, mask_ratio=mask_ratio
+            )
+
+        return embeddings, attention_masks, label_masks, ids_restore
+
+    
+# Copied from transformers.models.beit.modeling_beit.BeitPatchEmbeddings with BeitPatchEmbedding->TvltPixelPatchEmbeddings
```

Comment:
```suggestion
# Copied from transformers.models.beit.modeling_beit.BeitPatchEmbeddings
```
You can add "with Beit->Tvlt" for instance after the Copied from statement, but it will only have an impact on all code below the class name. So no need for that here